lsp-clients.el

```diff
@@ -97,13 +97,9 @@ finding the executable with variable `exec-path'."
   :risky t
   :type '(repeat string))
 
-(defun lsp-typescript-javascript-tsx-jsx-activate-p (filename mode)
-  "Check if the javascript-typescript language server should be enabled
-based on FILE-NAME and MAJOR-MODE"
-  (or (member mode '(typescript-mode typescript-tsx-mode js-mode js-jsx-mode js2-mode js2-jsx-mode rjsx-mode))
-      (and (eq major-mode 'web-mode)
-           (or (string-suffix-p ".tsx" filename t)
-               (string-suffix-p ".jsx" filename t)))))
+(defun lsp-typescript-javascript-tsx-jsx-activate-p (filename &optional _)
+  "Check if the javascript-typescript language server should be enabled based on FILENAME."
+  (not (null (string-match-p (rx (one-or-more char) "." (or "ts" "js") (opt "x") string-end) filename))))
```

Comment:
I think `(not (null ` is not needed.

Reply:
Cool, will remove it. I was following the implementation of `s-matches?` from `s.el`. The only thing to note is that `s-matches-p` does not return `t`/`nil`, but rather the index (if found) / `nil`.

Reply:
I guess it is a convention of the library to return t/nil for each predicate function (there is such convention in clojure) although they do not follow it everywhere.